src/App.tsx

```diff
@@ -1,223 +1,209 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
 import { AuthContext } from './components/Auth/AuthContext';
+import { TodoList } from './components/TodoList/TodoList';
+import { TodosFilter } from './components/TodosFilter/TodosFilter';
+import { Error } from './components/Error/Error';
+import {
+  createTodos,
+  deleteTodos,
+  getTodos,
+  updateTodos,
+} from './api/todos';
+import { Todo, TodoStatus } from './types/Todo';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todoFilter, setTodoFilter] = useState<TodoStatus>(TodoStatus.All);
+  const [error, setError] = useState('');
+  const [query, setQuery] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const completedTodos = todos.filter(todoItem => todoItem.completed);
+  const activeTodos = todos.filter(todoItem => !todoItem.completed);
+  const isCompletedAll = activeTodos.length === 0;
+
+  const filteredTodos = useMemo(() => {
+    switch (todoFilter) {
+      case TodoStatus.Active:
+        return todos.filter(item => !item.completed);
+
+      case TodoStatus.Completed:
+        return todos.filter(item => item.completed);
+
+      default:
+        return todos;
+    }
+  }, [todos, todoFilter]);
+
+  const handleChangeTodoFilter = (value: TodoStatus) => setTodoFilter(value);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
+    if (user) {
+      getTodos(user.id)
+        .then(todosFromServer => setTodos(todosFromServer));
+    }
+  }, [user]);
+
+  const createNewTodo = (event: React.FormEvent<HTMLFormElement>) => {
```

Comment:
You can try to use [useCallback](https://beta.reactjs.org/apis/react/useCallback) hook to memoize the functions.
It will help to optimize your page permofance